pkg/api/v1/defaults.go

```diff
@@ -112,9 +112,10 @@ func SetDefaults_Firmware(obj *Firmware) {
 func SetDefaults_VirtualMachineInstance(obj *VirtualMachineInstance) {
 	// FIXME we need proper validation and configurable defaulting instead of this
 	if _, exists := obj.Spec.Domain.Resources.Requests[v1.ResourceMemory]; !exists {
-		obj.Spec.Domain.Resources.Requests = v1.ResourceList{
-			v1.ResourceMemory: resource.MustParse("8192Ki"),
+		if obj.Spec.Domain.Resources.Requests == nil {
+			obj.Spec.Domain.Resources.Requests = v1.ResourceList{}
 		}
+		obj.Spec.Domain.Resources.Requests[v1.ResourceMemory] = resource.MustParse("8192Ki")
```

Comment:
oh wow, so we were clobbering the whole requests map if memory wasn't specified previously. good catch :+1: